kong/plugins/key-auth/handler.lua

```diff
@@ -90,13 +92,21 @@ local function do_authentication(conf)
     return nil, { status = 500, message = "Invalid plugin configuration" }
   end
 
+  if conf.validate_signature then
+    if type(conf.signature_names) ~= "table" then
+      kong.log.err("no conf.signature_names set, aborting plugin execution")
+      return nil, { status = 500, message = "Invalid plugin configuration" }
```

Comment:
It should be avoided to return HTTP 500 from a plugin if possible. Here, our posture is to heavily rely on schema validation to ensure that invalid plugin configurations are impossible at runtime.

If schema validation of the plugin's configuration cannot enforce such guarantees, a hard-coded default behavior in the plugin should be resilient (i.e. ignoring the feature altogether as if it was never configured, possibly with a warning log or other way to notify the operators).

In this case, let's try to rely more heavily on schema validation to ensure this type of mis-configuration is never encountered at runtime.